hdfs3/core.py

```diff
@@ -509,18 +507,24 @@ def read(self, length=None):
         return b''.join(buffers)
 
     def readline(self, chunksize=2**16, lineterminator='\n'):
-        """ Read a buffered line, text mode. """
-        lineterminator = ensure_string(lineterminator)
+        """ Return a line using buffered reading. 
+
+        Reads and caches chunksize bytes of data, and caches lines
+        locally. Subsequent readline calls deplete those lines until
+        empty, when a new chunk will be read. Mixing readline with
+        read is not recommended.
```

Comment:
Why is it not recommended.  Can you specify a bit more about the file pointer business?